### DISCUSSION TITLE
Order Placement - Delete/orders/{orderId}

### DISCUSSION THREAD

**Discussion by chiny123** [OpenWealth/OpenWealth#168] [Q&A] [answered]
**Submitted by: Michael Wüst  - SGKB
Date: 31.05.2023**

What http status code should we use, if we receive a Delete request for a trade that is in an order state not allowing a cancellation (e.g. executed)?
Do we return an http 202 and the TPP has to implicitly know by the order state that the cancellation request will not be further processed? Or do we return an http error? If the latter is the case, which http status code would be used?

**Comment by dustyush0sq** [comment]
A  potential option could be HTTP 409 / Conflict, however , HTTP 403 seems to be more common indeed

**Comment by TLusti** [comment]
@MichaelWuestSGKB can we close this?

**Comment by MichaelWuestSGKB** [comment]
@TLusti Yes, the proposed solution words for us.

**Comment by TLusti** [comment, marked as answer]
I would tend towards the handling via Error instead of “hoping” that the WealthTech understands the HTTP 202 response.
Looking at the Errors, it would most probably be a HTTP 403 – Forbidden, and then you might state in the commonErrorResponse part that the Delete of this resource is not allowed.
